libbpf-cargo/src/gen.rs

```diff
@@ -785,6 +785,7 @@ fn gen_skel_contents(_debug: bool, raw_obj_name: &str, obj_file_path: &Path) ->
         }}
 
         unsafe impl<'a> Send for {name}Skel<'a> {{}}
+        unsafe impl<'a> Sync for {name}Skel<'a> {{}}
```

Comment:
It would be great if you were to lay out your reasoning for this change. Also see [here](https://github.com/libbpf/libbpf-rs/blob/master/CONTRIBUTING.md#submitting-a-pull-request): "Include useful details in the commit messages". Thanks.

Reply:
Done

Reply:
Thanks! Long term we should probably make `libbpf_rs::OpenObject` and `libbpf_rs::skeleton::ObjectSkeletonConfig<'a>` `Sync` (or at least convince ourselves that it would be a bad idea), but this seems reasonable.